File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        stack=[]
        for i in tokens:
            if i =="+":
                a=stack.pop()
                b=stack.pop()
                c=int(a)+int(b)
                stack.append(c)

            elif i =="-":
                a=stack.pop()
                b=stack.pop()
                c=int(b)-int(a)
                stack.append(c)
            
            elif i =="/":
                a=stack.pop()
                b=stack.pop()
                c=int(float(b)/int(a))
                stack.append(c)

            elif i =="*":
                a=stack.pop()
                b=stack.pop()
                c=int(b)*int(a)
                stack.append(c)   

            else:
                stack.append(int(i))       

        return stack[-1]

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for(string ch: tokens){
            if (ch == "+"){
                int first=st.top();
                st.pop();
                int second= st.top();
                st.pop();
                int k = first+second;
                st.push(k);
            }  
            else if (ch == "/"){
                int a=st.top();
                st.pop();
                int b = st.top();
                st.pop();
                int c = b/a;
                st.push(c);
            }

            else if ( ch =="*"){
                int k = st.top();
                st.pop();
                int j =st.top();
                st.pop();
                int l = k*j;
                st.push(l);
            }
            else if ( ch=="-"){
                int u = st.top();
                st.pop();
                int v =st.top();
                st.pop();
                st.push(v-u);
            }
            else {
                st.push(stoi(ch));
            }   
                  }
            return ( st.top()) ;   
    }
};